### PR TITLE
Add jitpack support for SNAPSHOTS

### LINKED ISSUE #4819: Add jitpack.yml to build snapshots
Using https://jitpack.io/#broadinstitute/gatk for building SNAPSHOTS (independent on your jfrog repository, which has a expired date for snapshots - https://github.com/broadinstitute/gatk/issues/4565) is not working any longer due to the requirement of git-lfs (e.g., https://jitpack.io/com/github/broadinstitute/gatk/4.0.4.0/build.log)

It will be nice to add a `jitpack.yml` file (see https://jitpack.io/docs/BUILDING/#custom-commands) to install dependencies - this will be useful for downstream project depending on SNAPSHOTS.

### PR DESCRIPTION
**Pull request by magicDGS**
Closes #4819

### CODE REVIEW

**Comment by magicDGS** [comment]
@lbergelson - maybe you can have a look, as you commented in the original issue. As the commit isn't in the repo yet, I cannot test if it is working here. But here are the logs for the commit in the repo where I did the change: https://jitpack.io/com/github/bioinformagik/gatk/6270fa279/build.log

File: scripts/prepareJitpackEnvironment.sh [hunk 1]
```diff
@@ -0,0 +1,11 @@
+#!/bin/bash
```
[the remark this answers is not in the document]
Comment (magicDGS) [reply scripts/prepareJitpackEnvironment.sh]: Done

File: scripts/prepareJitpackEnvironment.sh [hunk 2]
```diff
@@ -0,0 +1,11 @@
...
+
+GIT_LFS_VERSION="1.2.1"
+GIT_LFS_LINK=https://github.com/github/git-lfs/releases/download/v${GIT_LFS_VERSION}/git-lfs-linux-amd64-${GIT_LFS_VERSION}.tar.gz
+GIT_LFS="git-lfs-${GIT_LFS_VERSION}/git-lfs"
+echo "Downloading and untarring git-lfs binary"
+wget -qO- $GIT_LFS_LINK | tar xvz
+
+echo "Fetching LFS files."
+$GIT_LFS install
+$GIT_LFS pull
```
[the remark this answers is not in the document]
Comment (magicDGS) [reply scripts/prepareJitpackEnvironment.sh]: Done

File: scripts/prepareJitpackEnvironment.sh [hunk 3]
```diff
@@ -0,0 +1,11 @@
+#!/bin/bash
+
+GIT_LFS_VERSION="1.2.1"
```
[the remark this answers is not in the document]
Comment (magicDGS) [reply scripts/prepareJitpackEnvironment.sh]: Done

File: jitpack.yml [hunk 4]
```diff
@@ -0,0 +1,4 @@
+jdk:
```
[the remark this answers is not in the document]
Comment (magicDGS) [reply jitpack.yml]: I will first push the next changes to test the build (can also be tested from the fork, as this branch is not part of this repo) and then try to add the comments (jitpack have very bad documentation for the yml, I extracted the info from https://jitpack.io/docs/BUILDING/#custom-commands)

[the remark this answers is not in the document]
Comment (magicDGS) [reply jitpack.yml]: Done (only adding a purpose comment) - let me know if you would like more info

**Comment by magicDGS** [comment]
@lbergelson - the last commit works. Check the logs and artifacts in this folder: https://jitpack.io/com/github/bioinformagik/gatk/336d389ecc605f4138226dddda41981f78278e09

**Comment by magicDGS** [comment]
Back to you @lbergelson - also, do you think that this will still work after #5112?

**Comment by lbergelson** [comment]
@magicDGS It looks like jitpack runs `./gradlew install` ( https://jitpack.io/docs/BUILDING/#gradle-projects ) which will create both the normal and testutils jars after that's split out.  So it should be fine.  You'll obviously have to change how you import the jars to include the test utils jar but there's no getting around that.

**Comment by lbergelson** [comment]
@magicDGS Thanks for doing this.

**Pull request merged**